authentik/core/api/sources.py

```diff
@@ -154,6 +155,17 @@ def user_settings(self, request: Request) -> Response:
             matching_sources.append(source_settings.validated_data)
         return Response(matching_sources)
 
+    def destroy(self, request: Request, *args, **kwargs):
+        """Prevent deletion of built-in sources"""
+        instance: Source = self.get_object()
+
+        if instance.managed == "goauthentik.io/sources/inbuilt":
```

Comment:
This should be a constant somewhere :eyes: (It's also used in `authentik/core/apps.py`)

Reply:
For the managed outpost, we define it in `authentik/outposts/apps.py`, but I quite like this approach of having it directly attached to the model where it makes sense to use it.

I'll let @BeryJu be the judge of whether we should continue this way to start putting stuff on models directly.